search.go

```diff
@@ -275,6 +276,12 @@ func (e *Entry) Unmarshal(i interface{}) (err error) {
 				return fmt.Errorf("ldap: could not parse value '%s' into int field", values[0])
 			}
 			fv.SetInt(intVal)
+		case time.Time:
```

Comment:
Can you please add `time.Time` to the list of supported types in Line 189?
https://github.com/go-ldap/ldap/blob/b8b015395be87c33f39a5104122d1dbaffe999ad/search.go#L188-L189

Reply:
@cpuschma done, also added `*ldap.DN` handling ;-)